components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```
@@ -0,0 +1,481 @@
+
+        if (holder == null || oldHolder == null) {
+            return new PublisherPropertyDTO[0];
+        }
+        List<PublisherPropertyDTO> updatedPropertyDTOs = new ArrayList<>();
+        for (PublisherPropertyDTO newPropertyDTO : holder.getPropertyDTOs()) {
+            if (StringUtils.isNotBlank(newPropertyDTO.getId()) && StringUtils.isNotBlank(newPropertyDTO.getValue())) {
+
+                PublisherPropertyDTO oldPropertyDTO = oldHolder.getPropertyDTO(newPropertyDTO.getId());
+                if (oldPropertyDTO == null || !oldPropertyDTO.getValue().equalsIgnoreCase(newPropertyDTO.getValue())) {
+                    updatedPropertyDTOs.add(newPropertyDTO);
+                }
+            }
+        }
+        return updatedPropertyDTOs.toArray(new PublisherPropertyDTO[0]);
+    }
+
+    /**
+     * Sets the base64 encoded secret value of the secret subscriber properties, if it has been updated.
+     * If there is an exception in encrypting, the secret will not be persisted.
+     *
+     * @param propertyDTOs list of subscriber properties
+     */
+    private PublisherPropertyDTO[] encryptUpdatedSecretProperties(PublisherPropertyDTO[] propertyDTOs) {
+
+        if (propertyDTOs == null) {
+            return new PublisherPropertyDTO[0];
+        }
+        List<PublisherPropertyDTO> updatedPropertyDTOs = new ArrayList<>();
+        for (PublisherPropertyDTO propertyDTO : propertyDTOs) {
+            if (propertyDTO.isSecret()) {
+                try {
+                    String encryptedValue = CryptoUtil.getDefaultCryptoUtil()
+                            .encryptAndBase64Encode(propertyDTO.getValue().getBytes());
+                    propertyDTO.setValue(encryptedValue);
+                } catch (CryptoException e) {
+                    LOG.error("Error while encrypting secret value of subscriber. Secret would not be persisted.", e);
+                    continue;
```

Throw an exception instead of continue.